src/mscorlib/shared/System/ArgumentException.cs

```diff
@@ -64,12 +65,13 @@ public ArgumentException(String message, String paramName)
         protected ArgumentException(SerializationInfo info, StreamingContext context)
             : base(info, context)
         {
-            throw new PlatformNotSupportedException();
+            _paramName = info.GetString("ParamName");
         }
 
         public override void GetObjectData(SerializationInfo info, StreamingContext context)
         {
             base.GetObjectData(info, context);
```

Comment:
desktop had a check for null info. I think you need this, if you want to throw ANE instead of NRE

Reply:
that's done in the base class ctor

Reply:
I notice it didn't have a similar check on the serialization constructor, though...

Reply:
just for reference: https://github.com/dotnet/coreclr/blob/master/src/mscorlib/src/System/Exception.cs#L74